front/lib/api/assistant/global_agents.ts

```diff
@@ -1046,10 +1047,15 @@ export async function getGlobalAgents(
     )
   );
 
+  const whiteListedProviders = owner.whiteListedProviders ?? MODEL_PROVIDER_IDS;
   const globalAgents: AgentConfigurationType[] = [];
 
   for (const agentFetcherResult of agentCandidates) {
-    if (agentFetcherResult) {
+    if (
+      agentFetcherResult &&
+      agentFetcherResult.scope === "global" &&
+      whiteListedProviders.includes(agentFetcherResult.model.providerId)
```

Comment:
Please use the function we introduced to do that check

Reply:
Forgot about it, my bad

Reply:
Also added it in another file performing the same check